packages/authorization-agent/src/authorization-agent.ts

```diff
@@ -130,4 +183,109 @@ export class AuthorizationAgent {
       affectedAuthorizations.map(async (accessAuthorization) => this.generateAccessGrant(accessAuthorization.iri))
     );
   }
+
+  public async findSocialAgentsWithAccess(dataInstanceIri: string): Promise<AgentWithAccess[]> {
```

Comment:
Break this large function in to two separate blocks for each loop. Also consider using `Promise.all` if possible.